x/leverage/keeper/oracle.go

```diff
@@ -39,9 +43,20 @@ func (k Keeper) TokenPrice(ctx sdk.Context, baseDenom string, mode types.PriceMo
 		if err != nil {
 			return sdk.ZeroDec(), t.Exponent, errors.Wrap(err, "oracle")
 		}
+		if !mode.AllowsExpired() {
```

Comment:
This is required for frontend. Right now, when a price goes down (for example, every day for 10 minutes during the osmosis epoch for some assets), users are seeing their positions values go to zero on the web app, which is not a good experience.

Even though this does not affect transactions (which require a non-expired price to function, both on mainnet and in this feature), allowing queries to use old prices instead of zero during outages is the improvement we've been requested to make.